fili-core/src/test/groovy/com/yahoo/bard/webservice/metadata/DataSourceMetadataLoadTaskSpec.groovy

```diff
@@ -24,121 +22,23 @@ import com.yahoo.bard.webservice.table.ConstrainedTable
 import com.yahoo.bard.webservice.table.StrictPhysicalTable
 import com.yahoo.bard.webservice.table.PhysicalTableDictionary
 
-import org.joda.time.DateTime
-import org.joda.time.DateTimeZone
 import org.joda.time.Interval
-import org.joda.time.format.DateTimeFormat
 
-import spock.lang.Shared
-import spock.lang.Specification
+class DataSourceMetadataLoadTaskSpec extends BaseDataSourceMetadataSpec {
 
-class DataSourceMetadataLoadTaskSpec extends Specification {
     private static final ObjectMappersSuite MAPPERS = new ObjectMappersSuite()
 
-    String tableName = TestDruidTableName.ALL_PETS.asName()
+    Interval interval1
+    Interval interval2
+    Interval interval3
 
-    Interval interval1 = Interval.parse("2015-01-01T00:00:00.000Z/2015-01-02T00:00:00.000Z")
-    Interval interval2 = Interval.parse("2015-01-02T00:00:00.000Z/2015-01-03T00:00:00.000Z")
-    Interval interval3 = Interval.parse("2015-01-03T00:00:00.000Z/2015-01-04T00:00:00.000Z")
-    Interval interval123 = Interval.parse("2015-01-01T00:00:00.000Z/2015-01-04T00:00:00.000Z")
+    List<String> dimensions13
```

Comment:
In general, one thing your strategy confuse me is that in the original Spec, we have something like:
```
List<String> dimensions123 = [dim1, dim2, dim3]*.asName()		
List<String> dimensions13 = [dim1, dim3]*.asName()
```
Your strategy choose to use `generateDimensions()` to populate your new `dimensions123` as the shared variable  `dimensions`, but for `dimensions13` it's populate in the `setup()`. It's a bit inconsistent for me that two similar variables in the old Spec get to be set in different ways. 

I might be wrong, but here is my suggestion:
1. Define all `generateBlah()` in `BaseDataSourceMetadataSpec` and for each `generateBlah()`, populate the map with all possible data you will be using in all child Spec, you can rename the keys to distinguish between which child Spec will be using it:
```
    Map<String, Interval> generateIntervals() {
        return [
                loadTaskSpecInterval1: Interval.parse("2015-01-01T00:00:00.000Z/2015-01-02T00:00:00.000Z"),
                loadTaskSpecInterval2: Interval.parse("2015-01-02T00:00:00.000Z/2015-01-03T00:00:00.000Z"),
                loadTaskSpecInterval3: Interval.parse("2015-01-01T00:00:00.000Z/2015-01-03T00:00:00.000Z"),
                SegmentSpecInterval1: Interval.parse("2015-01-01T00:00:00.000Z/2015-01-04T00:00:00.000Z"),
                ...
        ]
    }
```

2. Don't override any `generateBlah()` in your child Spec, then in your `setup()`:
```
def setup() {
        Interval loadTaskSpecInterval1 = intervals.loadTaskSpecInterval1
        Interval loadTaskSpecInterval2 = intervals.loadTaskSpecInterval2
        ....
```

With these change you can avoid calling those `generateBlah()` functions over and over again in your `childSetupSepc()` as mentioned by @archolewa . You don't even need a `childSetupSepc()` because the parent Spec does all the population for you.  😼

Reply:
`dimensions123` is gone. `dimensions` serves as a resource container for all dimensions that are to be used in a Spec

The reason for not generating these resources but having child extend and override is that, for example, if you override `generateInterval`, `generateSegment` throws error in other Specs, since segments depends on generated intervals

If we initialize all the resources  in the way you recommended, we didn't solve the old problem because we we just changing variable into strings.

Reply:
Sounds reasonable